core/src/main/scala/pureconfig/package.scala

```diff
@@ -143,9 +166,13 @@ package object pureconfig {
    */
   @throws[ConfigReaderException[_]]
   def loadConfigOrThrow[Config](path: Path)(implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
-    ConfigFactory.invalidateCaches()
-    val config = ConfigFactory.load(ConfigFactory.parseFile(path.toFile))
-    getResultOrThrow[Config](loadConfig[Config](config)(conv))
+    val errorOrConfig =
+      for {
+        _ <- invalidateCaches().right
+        rawConfig <- loadFile(path).right
+        config <- loadConfig[Config](rawConfig)(conv).right
+      } yield config
+    getResultOrThrow[Config](errorOrConfig)
```

Comment:
I was thinking the same and I was wondering why we didn't implemented the `loadConfigOrThrow` by using `loadConfig`. Then I noticed that the location is filled differently, e.g.

```scala
  def loadConfig[Config](namespace: String)(implicit conv: ConfigConvert[Config]): Either[ConfigReaderFailures, Config] = {
    for {
      _ <- invalidateCaches().right
      rawConfig <- load().right
      config <- improveFailures[Config](
        loadConfig[Config](rawConfig.getConfig(namespace))(conv), namespace, None).right
    } yield config
  }
```

but

```scala
def loadConfigOrThrow[Config](namespace: String)(implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
    val errorOrConfig =
      for {
        _ <- invalidateCaches().right
        rawConfig <- load().right.map(_.getConfig(namespace)).right
        config <- improveFailures[Config](
          loadConfig[Config](rawConfig)(conv), namespace, ConfigValueLocation(rawConfig.root())).right
      } yield config
    getResultOrThrow[Config](errorOrConfig)
  }
```

I'm not sure why the first set the position of the error to `None` while the second to `ConfigValueLocation(rawConfig.root())` but this is preventing me from reusing the first in the second. @jcazevedo do you know why the two methods are different for the error location?